Fails to build with musl-gcc w/ linux-lts headers (no `struct clone_args`)

I install musl-gcc and kernel-headers-musl on Arch Linux, then create a switch with `CC=musl-gcc opam switch create ...`. Installing `eio_linux` fails as follows:

```
File "lib_eio_linux/dune", line 13, characters 9-18:
13 |   (names eio_stubs))
              ^^^^^^^^^
(cd _build/default/lib_eio_linux && /usr/bin/musl-gcc -O2 -fno-strict-aliasing -fwrapv -pthread -fPIC -pthread -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -g -I /home/fabian/.opam/5.2.0+musl/lib/ocaml -I /home/fabian/.opam/5.2.0+musl/lib/bigstringaf -I /home/fabian/.opam/5.2.0+musl/lib/cstruct -I /home/fabian/.opam/5.2.0+musl/lib/domain-local-await -I /home/fabian/.opam/5.2.0+musl/lib/eio -I /home/fabian/.opam/5.2.0+musl/lib/eio/core -I /home/fabian/.opam/5.2.0+musl/lib/eio/unix -I /home/fabian/.opam/5.2.0+musl/lib/eio/utils -I /home/fabian/.opam/5.2.0+musl/lib/fmt -I /home/fabian/.opam/5.2.0+musl/lib/hmap -I /home/fabian/.opam/5.2.0+musl/lib/logs -I /home/fabian/.opam/5.2.0+musl/lib/lwt-dllist -I /home/fabian/.opam/5.2.0+musl/lib/mtime -I /home/fabian/.opam/5.2.0+musl/lib/mtime/clock/os -I /home/fabian/.opam/5.2.0+musl/lib/ocaml/threads -I /home/fabian/.opam/5.2.0+musl/lib/ocaml/unix -I /home/fabian/.opam/5.2.0+musl/lib/optint -I /home/fabian/.opam/5.2.0+musl/lib/psq -I /home/fabian/.opam/5.2.0+musl/lib/seq -I /home/fabian/.opam/5.2.0+musl/lib/thread-table -I /home/fabian/.opam/5.2.0+musl/lib/uring -I ../lib_eio/unix/include -o eio_stubs.o -c eio_stubs.c)
eio_stubs.c:161:42: warning: ‘struct clone_args’ declared inside parameter list will not be visible outside of this definition or declaration
  161 | static pid_t clone3_with_fallback(struct clone_args *cl_args) {
      |                                          ^~~~~~~~~~
eio_stubs.c: In function ‘clone3_with_fallback’:
eio_stubs.c:162:42: error: invalid use of undefined type ‘struct clone_args’
  162 |   int *pidfd = (int *)(uintptr_t) cl_args->pidfd;
      |                                          ^~
...
```

Note that the kernel-headers-musl package uses 4.19.88 (LTS) headers, which don't have `struct clone_args`: https://github.com/sabotage-linux/kernel-headers

--- a/lib_eio_linux/eio_stubs.c
+++ b/lib_eio_linux/eio_stubs.c
@@ -39,7 +39,11 @@
 #ifndef SYS_clone3
 # define SYS_clone3 435
 # define CLONE_PIDFD 0x00001000
-struct clone_args {
+#endif
+
+// struct clone_args isn't defined in linux-lts headers, so define it here
+// Note that this struct is versioned by size. See linux/sched.h for details
+struct caml_eio_clone_args {
   uint64_t flags;
   uint64_t pidfd;
   uint64_t child_tid;
@@ -48,11 +52,7 @@ struct clone_args {
   uint64_t stack;
   uint64_t stack_size;
   uint64_t tls;
-  uint64_t set_tid;
-  uint64_t set_tid_size;
-  uint64_t cgroup;
 };
-#endif
 
 // Make sure we have enough space for at least one entry.
 #define DIRENT_BUF_SIZE (PATH_MAX + sizeof(struct dirent64))
@@ -178,9 +178,9 @@ static int pidfd_open(pid_t pid, unsigned int flags) {
 
 /* Like clone3, but falls back to fork if not supported.
    Also, raises exceptions rather then returning an error. */
-static pid_t clone3_with_fallback(struct clone_args *cl_args) {
+static pid_t clone3_with_fallback(struct caml_eio_clone_args *cl_args) {
   int *pidfd = (int *)(uintptr_t) cl_args->pidfd;
-  pid_t child_pid = syscall(SYS_clone3, cl_args, sizeof(struct clone_args));
+  pid_t child_pid = syscall(SYS_clone3, cl_args, sizeof(struct caml_eio_clone_args));
 
   if (child_pid >= 0)
     return child_pid;		/* Success! */
@@ -216,7 +216,7 @@ CAMLprim value caml_eio_clone3(value v_errors, value v_actions) {
   CAMLlocal1(v_result);
   pid_t child_pid;
   int pidfd = -1;		/* Is automatically close-on-exec */
-  struct clone_args cl_args = {
+  struct caml_eio_clone_args cl_args = {
     .flags = CLONE_PIDFD,
     .pidfd = (uintptr_t) &pidfd,
     .exit_signal = SIGCHLD,	/* Needed for wait4 to work if we exit before exec */

